packages/ckeditor5-link/tests/linkimageediting.js

```diff
@@ -711,4 +711,155 @@ describe( 'LinkImageEditing', () => {
+								label: 'Downloadable',
+								attributes: {
+									download: 'download'
+								}
+							},
+							isGallery: {
+								mode: 'manual',
+								label: 'Gallery link',
+								attributes: {
+									class: 'gallery'
+								}
+							}
+						}
+					}
+				} )
+				.then( newEditor => {
+					editor = newEditor;
+					model = editor.model;
+				} );
+		} );
+
+		afterEach( () => {
+			return editor.destroy();
+		} );
+
+		// See: #7975.
+		describe( 'manual decorators: linked text and linked image', () => {
+			it( 'should upcast the decorators when linked image (figure > a > img)', () => {
+				editor.setData(
+					'<figure class="image">' +
+						'<a class="gallery" href="https://cksource.com" target="_blank" rel="noopener noreferrer" download="download">' +
+							'<img src="sample.jpg" alt="bar">' +
+						'</a>' +
+						'<figcaption>Caption</figcaption>' +
+					'</figure>' +
+					'<p>' +
+						'<a href="https://cksource.com" target="_blank" rel="noopener noreferrer" download="download">' +
+							'https://cksource.com' +
+						'</a>' +
+					'</p>'
+				);
+
+				expect( getModelData( model, { withoutSelection: true } ) ).to.equal(
+					'<image alt="bar" ' +
+						'linkHref="https://cksource.com" ' +
+						'linkIsDownloadable="true" ' +
+						'linkIsExternal="true" ' +
+						'linkIsGallery="true" ' +
+						'src="sample.jpg">' +
+					'</image>' +
+					'<paragraph>' +
+						'<$text linkHref="https://cksource.com" linkIsDownloadable="true" linkIsExternal="true">' +
+							'https://cksource.com' +
+						'</$text>' +
+					'</paragraph>'
+				);
+			} );
+
+			it( 'should upcast the decorators when linked image (a > img)', () => {
+				editor.setData(
+					'<a class="gallery" href="https://cksource.com" target="_blank" rel="noopener noreferrer" download="download">' +
+						'<img src="sample.jpg" alt="bar">' +
+					'</a>' +
+					'<p>' +
+						'<a href="https://cksource.com" target="_blank" rel="noopener noreferrer" download="download">' +
+							'https://cksource.com' +
+						'</a>' +
+					'</p>'
+				);
+
+				expect( getModelData( model, { withoutSelection: true } ) ).to.equal(
+					'<image alt="bar" ' +
+						'linkHref="https://cksource.com" ' +
+						'linkIsDownloadable="true" ' +
+						'linkIsExternal="true" ' +
+						'linkIsGallery="true" ' +
+						'src="sample.jpg">' +
+					'</image>' +
+					'<paragraph>' +
+						'<$text linkHref="https://cksource.com" linkIsDownloadable="true" linkIsExternal="true">' +
+							'https://cksource.com' +
+						'</$text>' +
+					'</paragraph>'
+				);
+			} );
+
+			it( 'should downcast the decorators after applying a change', () => {
+				setModelData( model,
+					'<image alt="bar" src="sample.jpg"></image>' +
+					'<paragraph>' +
+						'<$text>https://cksource.com</$text>' +
+					'</paragraph>'
+				);
+
+				model.change( writer => {
+					// <image>
+					writer.setSelection( model.document.getRoot().getChild( 0 ), 'on' );
```

Comment:
A much simpler approach is to select image by enclosing it in brackets in line

https://github.com/ckeditor/ckeditor5/blob/dec11ed084ca8670569075d0a697f35a2997536d/packages/ckeditor5-link/tests/linkimageediting.js#L822

Like so `'[<image alt="bar" src="sample.jpg"></image>]' +`.

Reply:
On a side note: I checked that the image does not necessarily need a focus, as the test passes if I remove this particular `model.change()` call (I assume that editor will simply put the focus to the first element by default).